extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/vertx/VertxHttpServerMetrics.java

```diff
@@ -62,6 +62,7 @@ public class VertxHttpServerMetrics extends VertxTcpServerMetrics
         activeRequests = new LongAdder();
         Gauge.builder(config.getHttpServerActiveRequestsName(), activeRequests, LongAdder::doubleValue)
                 .tag("url.scheme", httpServerOptions.isSsl() ? "https" : "http")
+                .tag("server.port", "" + httpServerOptions.getPort())
```

Comment:
It won't work for random port and mapped port (containers).

I think we need to be slightly more careful here. We can have the scheme, but the public port is unknown.

Reply:
Does it really matter if it's the public port or not? Essentially this would reflect the configuration values.
The only situation it wouldn't work properly is when all ports are configured to be random (which I doubt is a common scenario)

Reply:
We will only now that properly when we have a service registry with runtime values.
I think we can do this for now because production values will match.

Reply:
Ok, if we ignore the random ports. I still think it should report the public port and not the private port, but as I said, you cannot have that value (typically on OpenShift it would mean finding the service representing you and extracting the port - way to expensive and specific). 

Maybe we should not add this tag if the port is 0, or negative.

Reply:
> you cannot have that value

Yeah exactly.

> Maybe we should not add this tag if the port is 0, or negative.

Makes sense.

Reply:
PR updated